### PR TITLE
Added pre-commit job for mariadb operator

### PR DESCRIPTION
**Pull request by amolkahat**
This patch adds pre-commit job for mariadb operator.

### CODE REVIEW

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

File: ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml
Comment (viroel) [inline ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml:1]: By adding pre-commit job, we should also remove duplicated jobs: govet, gofmt, golint will already be tested by pre-commit job.

**Review by viroel — CHANGES_REQUESTED** [review]

File: ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml
Comment (amolkahat) [reply ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml:1]: I agree with you. But we can't remove those until we are adding pre-commit jobs.

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

File: ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml [hunk 1]
```diff
@@ -97,6 +102,17 @@ tests:
...
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAGS=
+    export PRE_COMMIT_HOME=/tmp
+    export GOLANGCI_LINT_CACHE=/tmp
+    mkdir -p ../operator/ && cp -r . ../operator
+    cd ../operator
+    pre-commit run -a --show-diff-on-failure
+  container:
+    from: openstack-operator-ci-pre-commit
```
Comment (marios) [inline ci-operator/config/openstack-k8s-operators/mariadb-operator/openstack-k8s-operators-mariadb-operator-master.yaml:115]: sanity check - looks like the repo already has the .pre-commit-config https://github.com/openstack-k8s-operators/mariadb-operator/blob/9b37cd889f189f86af017b8b0e51dd28bbcc7127/.pre-commit-config.yaml#L1

**Review by marios — CHANGES_REQUESTED** [review]
looks right but please update the commit message s/maradb/mariadb

**Review by marios — APPROVED** [review]
lgtm 

no precommit job run on the latest updates but that should be OK since it was just a commit message update

run history there https://prow.ci.openshift.org/job-history/gs/origin-ci-test/pr-logs/directory/rehearse-36011-pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

last result from 28th @ https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/36011/rehearse-36011-pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check/1640612596358844416/build-log.txt

**Comment by viroel** [comment]
@amolkahat hi, can you remove jobs that are now duplicated, e.g. govet, gofmt, since they run in pre-commit.

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

**Comment by bshewale** [comment]
Will need this PR first to merge : https://github.com/openstack-k8s-operators/mariadb-operator/pull/90

**Comment by viroel** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-mariadb-operator-master-precommit-check

**Pull request merged**